app/models/manageiq/providers/openstack/identity_sync_mixin.rb

```diff
@@ -0,0 +1,180 @@
+module ManageIQ::Providers::Openstack::IdentitySyncMixin
+  def list_users
+    # V2 requires adminURL as endpoint_type
+    connection_options = {:service => "Identity", :openstack_endpoint_type => 'adminURL'}
+    ext_management_system.with_provider_connection(connection_options) do |service|
+      service.list_users.body["users"]
+    end
+  end
```

Comment:
Would keystone.users.all also work? The identity delegate code fore creating tenants uses that for listing all users, although it does so differently for keystone v2 and v3. Maybe it makes sense to create a list_users, list_users_v2, and list_users_v3 in the delegate (and also put the create methods in there as well)?

Reply:
Nevermind about the create users, I just realized those are ManageIQ users being created, so that can't go in identity_delegate. I think list_users may belong there, though.

Reply:
I agree that list_users looks out of place in identity_sync_mixin. But it ddidn't work when I tried to put an equivalent method in identity_delegate like:

    def list_all_users
      # list_users fails in V2, because the URI is invalid
      # V2 requires adminURL as endpoint_type
      connection_options = {:service => "Identity", :openstack_endpoint_type => 'adminURL'}
      connection = @os_handle.connect(connection_options)
      connection.list_users.body["users"]
    end

will result in 

Expected([200, 204]) <=> Actual(404 Not Found) excon.error.response :body => "{\"error\": {\"message\": \"The resource could not be found.\", \"code\": 404, \"title\": \"Not Found\"}}" :cookies => [ ] :headers => { "Content-Length" => "93" "Content-Type" => "application/json" "Date" => "Wed, 25 Oct 2017 20:25:04 GMT" "Server" => "Apache" "Vary" => "X-Auth-Token" "x-openstack-request-id" => "req-695652d6-2cf0-496a-81a8-e502f148624f" } :host => "192.168.24.11" :local_address => "192.168.122.1" :local_port => 41668 :path => "/v2.0//users" :port => 5000 :reason_phrase => "Not Found" :remote_ip => "192.168.24.11" :status => 404 :status_line => "HTTP/1.1 404 Not Found\r\n" [ems_cloud/sync_users]